.github/workflows/ci.yaml

```diff
@@ -228,6 +228,86 @@ jobs:
         with:
           files: lcov.info
 
+  build-and-test-minimal:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set rust nightly version
+        run: echo "NIGHTLY_VERSION=$(curl https://rust-lang.github.io/rustup-components-history/x86_64-unknown-linux-gnu/rustc)" >> $GITHUB_ENV
+
+      - name: Install rust
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: nightly-${{ env.NIGHTLY_VERSION }}
+          override: true
+
+      - name: Install dependencies
+        uses: awalsh128/cache-apt-pkgs-action@latest
+        with:
+          packages: libpam0g-dev
+          version: "1.0"
+
+      - name: Rust Cache
+        uses: Swatinem/rust-cache@v2
+        with:
+          shared-key: "nightly"
+
+      - name: Update to minimal direct dependencies
+        uses: actions-rs/cargo@v1
+        with:
+          command: update
+          args: -Zdirect-minimal-versions
```

Comment:
Normally we still run with the versions in our `Cargo.lock`, but we have lower versions specified in `Cargo.toml`. This nightly only flag basically runs the cargo resolver in such a way that the lowest compatible direct dependencies are the ones that will be picked instead of the latest compatible version. In such a way we can test if we still compile with the versions in our `Cargo.toml` and those do not get outdated because we use some new library feature. This is mainly useful for packagers, which might not use our `Cargo.lock` and might instead look at what is available in their own OS repository instead. Technically we'd have to check any in-between version as well, but I think testing against the versions in our lockfile and the minimal versions is good enough for now.

Reply:
This is true for any piece of software on any typical linux distro (or bsd for that matter, probably just about anything except for nix). Similar to how in the C world you almost never depend on a specific version of a specific dependency. Of course maintainers of distro packages patch security vulnerabilities when they arise by backporting fixes to their own versions (or more likely upgrading in the case of something like arch linux).

As for us depending on vulnerable packages: I'd say that we would always recommend to use the latest semver compatible versions of our dependencies (and those are the only ones we would support for our self-distributed releases), as defined in our lockfile. But our toml file should indicate which versions of our dependencies have all the functionality we need, regardless of security status.